docs/user/build-customization.rst

```diff
@@ -33,60 +33,150 @@ The jobs where users can customize our default build process are:
-* Each command is executed in a new shell process, so modifications done to the shell environment do not persist between commands
+* The current working directory is at the root of your project's cloned repository.
+* Environment variables are expanded for each individual command (see :doc:`/reference/environment-variables`).
+* Each command is executed in a new shell process, so modifications done to the shell environment do not persist between commands.
 * Any command returning non-zero exit code will cause the build to fail immediately
-  (note there is a special exit code to `cancel the build <cancel-build-based-on-a-condition>`_)
-* ``build.os`` and ``build.tools`` are required when using ``build.jobs``
+  (note there is a special exit code to `cancel the build <cancel-build-based-on-a-condition>`_).
+* ``build.os`` and ``build.tools`` are required when using ``build.jobs``.
+* If the :ref:`config-file/v2:sphinx` or :ref:`config-file/v2:mkdocs` configuration is defined,
+  the ``create_environment``, ``install``, and ``build`` jobs will use the default commands for the selected tool.
+* If neither of the :ref:`config-file/v2:sphinx` or :ref:`config-file/v2:mkdocs` configurations are defined,
+  the ``create_environment``, ``install``, and ``build`` jobs will default to run nothing,
+  giving you full control over the build process.
+
+Where to put files
+~~~~~~~~~~~~~~~~~~
+
+It is your responsibility to generate HTML and other formats of your documentation when overriding the steps from :ref:`config-file/v2:build.jobs.build`.
+The contents of the ``$READTHEDOCS_OUTPUT/<format>/`` directory will be hosted as part of your documentation.
+
+We store the the base folder name ``_readthedocs/`` in the environment variable ``$READTHEDOCS_OUTPUT`` and encourage that you use this to generate paths.
+
+Supported :ref:`formats <downloadable-documentation:accessing offline formats>` are published if they exist in the following directories:
+
+* ``$READTHEDOCS_OUTPUT/html/`` (required)
+* ``$READTHEDOCS_OUTPUT/htmlzip/``
+* ``$READTHEDOCS_OUTPUT/pdf/``
+* ``$READTHEDOCS_OUTPUT/epub/``
 
+.. note::
 
-``build.jobs`` examples
-~~~~~~~~~~~~~~~~~~~~~~~
+   Remember to create the folders before adding content to them.
+   You can ensure that the output folder exists by adding the following command:
+
+   .. code-block:: console
+
+       mkdir -p $READTHEDOCS_OUTPUT/html/
+
+Search support
+~~~~~~~~~~~~~~
+
+Read the Docs will automatically index the content of all your HTML files,
+respecting the :ref:`search <config-file/v2:search>` option.
+
+You can access the search from the Read the Docs :term:`dashboard`,
+or by using the :doc:`/server-side-search/api`.
+
+.. note::
+
+   In order for Read the Docs to index your HTML files correctly,
+   they should follow the conventions described at :doc:`rtd-dev:search-integration`.
+
+build.commands
+~~~~~~~~~~~~~~
```

Comment:
I would call this "Completely override the build process" as it is presented at the beginning of the page. Also, I'd use `----` for the section:

```suggestion
Completely override the build process
-----------------------------------------------------------
```

Reply:
I would avoid having a title with just `build.commands` since it doesn't communicate too much the users.

Reply:
I think this should under this section, but renamed it to "alternative syntax"